Compilation error: Feature 'declaration expression' cannot be used

I downloaded the latest source (1.2.5) zip, and tried compiling it using `monobuild.cmd` under Windows Server with latest mono stable (5.0.1.1), but it throws an error during compilation:

> Building StackExchange.Redis.dll ...
> StackExchange.Redis\StackExchange\Redis\CommandMap.cs(179,49): error CS1644: Feature `declaration expression' cannot be used because it is not part of the C# 6.0 language specification
> Compilation failed: 1 error(s), 0 warnings

Any clues why this would occur, and how to resolve it?

**Edit:** Same error when compiling it using `monobuild.bash` under Ubuntu Server 16.04 with latest mono stable (5.0.1.1).

--- a/monobuild.bash
+++ b/monobuild.bash
@@ -3,9 +3,9 @@ rm -rf BasicTest/bin/mono
 mkdir -p StackExchange.Redis/bin/mono
 mkdir -p BasicTest/bin/mono
 echo -e "Building StackExchange.Redis.dll ..."
-mcs -recurse:StackExchange.Redis/*.cs -out:StackExchange.Redis/bin/mono/StackExchange.Redis.dll -target:library -unsafe+ -o+ -r:System.IO.Compression.dll
+csc -recurse:StackExchange.Redis/*.cs -out:StackExchange.Redis/bin/mono/StackExchange.Redis.dll -target:library -unsafe+ -o+ -r:System.IO.Compression.dll
 echo -e "Building BasicTest.exe ..."
-mcs BasicTest/Program.cs -out:BasicTest/bin/mono/BasicTest.exe -target:exe -o+ -r:StackExchange.Redis/bin/mono/StackExchange.Redis.dll
+csc BasicTest/Program.cs -out:BasicTest/bin/mono/BasicTest.exe -target:exe -o+ -r:StackExchange.Redis/bin/mono/StackExchange.Redis.dll
 cp StackExchange.Redis/bin/mono/*.* BasicTest/bin/mono/
 echo -e "Running basic test ..."
 mono BasicTest/bin/mono/BasicTest.exe 10000

--- a/monobuild.cmd
+++ b/monobuild.cmd
@@ -3,9 +3,9 @@
 @md StackExchange.Redis\bin\mono 1>nul 2>nul
 @md BasicTest\bin\mono 1>nul 2>nul
 @echo Building StackExchange.Redis.dll ...
-@call mcs -recurse:StackExchange.Redis\*.cs -out:StackExchange.Redis\bin\mono\StackExchange.Redis.dll -target:library -unsafe+ -o+ -r:System.IO.Compression.dll
+@call csc -recurse:StackExchange.Redis\*.cs -out:StackExchange.Redis\bin\mono\StackExchange.Redis.dll -target:library -unsafe+ -o+ -r:System.IO.Compression.dll
 @echo Building BasicTest.exe ...
-@call mcs BasicTest\Program.cs -out:BasicTest\bin\mono\BasicTest.exe -target:exe -o+ -r:StackExchange.Redis\bin\mono\StackExchange.Redis.dll
+@call csc BasicTest\Program.cs -out:BasicTest\bin\mono\BasicTest.exe -target:exe -o+ -r:StackExchange.Redis\bin\mono\StackExchange.Redis.dll
 @copy StackExchange.Redis\bin\mono\*.* BasicTest\bin\mono > nul
 @echo .
 @echo Running basic test (Mono) ...

